waiter/src/waiter/core.clj

```diff
@@ -163,14 +163,36 @@
                   (cid/ensure-correlation-id nested-response get-request-cid)
                   nested-response)))))))))
 
-(defn wrap-support-info
-  "Attaches support-info to the request."
-  [handler support-info]
-  (fn wrap-support-info-fn [request]
+(defn wrap-request-info
+  "Attaches request info to the request."
+  [handler router-id support-info]
+  (fn wrap-request-info-fn [request]
     (-> request
-        (assoc :support-info support-info)
+        (assoc :request-id (str (utils/unique-identifier) "-" (-> request utils/request->scheme name))
+               :request-time (t/now)
+               :router-id router-id
+               :support-info support-info)
         handler)))
 
+(defn wrap-debug
+  "Attaches debugging headers to requests when enabled."
+  [handler]
+  (fn wrap-debug-fn
+    [{:keys [request-id request-time router-id] :as request}]
+    (if (utils/request->debug-enabled? request)
+      (let [response (handler request)
+            add-headers (fn [response]
+                          (update response :headers
+                                  (fn [headers]
+                                    (-> headers
+                                        (assoc "X-Waiter-Request-Date" (utils/date-to-str request-time utils/formatter-rfc822))
+                                        (assoc "X-Waiter-Request-Id" request-id)
+                                        (assoc "X-Waiter-Router-Id" router-id)))))]
+        (if (au/chan? response)
```

Comment:
Should we create a helper function for this pattern in `middleware.clj` where we apply a function to the response?